From b0a0f5232634cb0e598adc3947135be3b72b11cf Mon Sep 17 00:00:00 2001
From: Niko Tyni <ntyni@debian.org>
Date: Sat, 25 Feb 2012 19:41:27 +0200
Subject: Make EU::MM pass LD through to recursive Makefile.PL invocations

In a directory hierarchy with Makefile.PL files in subdirectories,
command line arguments like
 perl Makefile.PL LD="ld -Wl,-z relro"
are not used when descending into the subdirectories.

This seems to be by design: there's a short list of variables
that are passed through when invoking 'make' in the subdirectories,
but the rest just use their default values.

Debian needs to pass LD through for sane handling of security related
build flags, so add that to the whitelist.

Bug: http://rt.cpan.org/Public/Bug/Display.html?id=28632
Bug-Debian: http://bugs.debian.org/660195
Patch-Name: debian/makemaker-pasthru.diff
---
 cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm b/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm
index 24a80e9..7d88702 100644
--- a/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm
+++ b/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm
@@ -2672,7 +2672,7 @@ sub pasthru {
     my($sep) = $Is{VMS} ? ',' : '';
     $sep .= "\\\n\t";
 
-    foreach my $key (qw(LIB LIBPERL_A LINKTYPE OPTIMIZE
+    foreach my $key (qw(LIB LIBPERL_A LINKTYPE OPTIMIZE LD
                      PREFIX INSTALL_BASE)
                  ) 
     {
